lib/util/http/host-rules.ts

```diff
@@ -68,46 +68,59 @@ function findMatchingRules(options: GotOptions, url: string): HostRule {
 }
 
 // Apply host rules to requests
-export function applyHostRules(url: string, inOptions: GotOptions): GotOptions {
-  const options: GotOptions = { ...inOptions };
+export function applyHostRules(
+  url: string,
+  inOptions: HttpOptions
+): HttpOptions['gotOptions'] {
+  const options: HttpOptions = { ...inOptions };
   const foundRules = findMatchingRules(options, url);
   const { username, password, token, enabled, authType } = foundRules;
-  if (options.noAuth) {
+  if (options?.gotOptions?.noAuth) {
     logger.trace({ url }, `Authorization disabled`);
   } else if (
-    options.headers?.authorization ||
-    options.password ||
-    options.token
+    options?.gotOptions?.headers?.authorization ||
+    options?.gotOptions?.password ||
+    options?.gotOptions?.token
```

Comment:
I think those fields should stay at `HttpOptions` here and later only relevant got options should be passed to got call

Reply:
you mean that I need to check `httpOptions` "headers, password, token"?
because the previous behavior was checking `GotOption` "headers, password, token" fields